### DISCUSSION TITLE
Why Javascript instead of Rust?

### DISCUSSION THREAD

**Discussion by azzamsa** [stoatchat/discussions#119] [General]
Hi.

I am wondering the reason behind preact (JS framework) instead of Yew/Seed/mzoon/sycamore/.. (Rust framework).
Since the backend is already written in Rust, I assume that the developers already love Rust's capabilities and features.

I don't want to start a heated discussion here, nor I don't hope that the revolt client will be rewritten in Rust.
I just want the reason, because I want to build a semi-production frontend app, and I am still confused about what to choose.
a Rust front-end framework, or a Javascript one.

Thanks a lot for Revolt.

Moved from https://github.com/revoltchat/revite/issues/235

**Comment by bdbch** [comment]
I guess because the entrypoint for Javascript is really low and almost all frontend developers work with Javascript / React.
